test/equality_test.dart

```diff
@@ -175,6 +175,25 @@ main() {
     expect(equality.hash("foo"), isNot(equals(equality.hash("bar"))));
     expect(equality.hash("fÕÕ"), isNot(equals(equality.hash("fõõ"))));
   });
+
+  test("EqualityBy", () {
+    var equality = new EqualityBy<NonCanonicalElement, String>((e) => e.id);
+    expect(
+        equality.equals(
+            new NonCanonicalElement("foo"), new NonCanonicalElement("foo")),
+        isTrue);
+    expect(
+        equality.equals(
+            new NonCanonicalElement("foo"), new NonCanonicalElement("bar")),
+        isFalse);
```

Comment:
Test `hash()` and `isValidKey()` as well (probably as separate tests in the same group.